src/test/java/bio/terra/app/controller/DatasetValidationsTest.java

```diff
@@ -146,20 +146,6 @@ public void testDuplicateColumnNames() throws Exception {
                 "InvalidAssetTableColumn", "InvalidAssetTableColumn", "InvalidRootColumn"});
     }
 
-    @Test
-    public void testMissingAssets() throws Exception {
-        DatasetRequestModel req = buildDatasetRequest();
-        req.getSchema().assets(null);
-        ErrorModel errorModel = expectBadDatasetCreateRequest(req);
-        checkValidationErrorModel("missingAssets", errorModel,
-            new String[]{"NotNull"});
-
-        req.getSchema().assets(Collections.emptyList());
-        errorModel = expectBadDatasetCreateRequest(req);
-        checkValidationErrorModel("missingAssets", errorModel,
-            new String[]{"NoAssets"});
-    }
```

Comment:
I tried to replace this with a positive check, but started hitting errors about missing tokens etc. It looked like the backend might have actually been trying to create datasets when the request was valid. I looked into what it'd take to extract the validation logic into a more isolated space for unit testing without mocks, and it seemed like more work than I'd want to do as part of unblocking us here.

Reply:
It is, we need a way to mock out the job service. I'll make a ticket, thanks for the fix!